crates/oxc_linter/src/rules/eslint/no_ternary.rs

```diff
@@ -0,0 +1,67 @@
+use oxc_ast::{ast::Expression,AstKind};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-ternary): Unexpected use of ternary expression")]
+#[diagnostic(severity(warning), help("Do not use the ternary expression."))]
+struct NoTernaryDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoTernary;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow ternary operators
+    ///
+    /// ### Why is this bad?
+    /// The ternary operator is used to conditionally assign a value to a variable. Some believe that the use of ternary operators leads to unclear code.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var foo = isBar ? baz : qux;
+    //
+    // function quux() {
+    //   return foo ? bar() : baz();
+    // }
+    /// ```
+    NoTernary,
+    style
+);
+
+impl Rule for NoTernary {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::ConditionalExpression(cond_expr) = node.kind() {
+            if cond_expr.consequent.is_binaryish()
+                || cond_expr.alternate.is_binaryish()
+                || cond_expr.test.is_binaryish()
+            {
+                ctx.diagnostic(NoTernaryDiagnostic(Span::new(
+                    cond_expr.span.start,
+                    cond_expr.span.end,
+                )));
+            }
+        }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![r#""x ? y";"#, "if (true) { thing() } else { stuff() };"];
```

Comment:
@mysteryven I have a problem with this first test, because is as rulegen created it, but seems strange to my, and I don't know how to pass it, I tried adding this test to other places where ternaries are checked and it fails. The second test was added by me, let me know if there is some problem. Thanks! 
https://github.com/eslint/eslint/blob/main/tests/lib/rules/no-ternary.js

Reply:
Ops, maybe there are some broken in our rulegen, we can change it to `let pass = vec![r#""x ? y";"#];`. 

It's a bit strange, I run `just new-rule no-ternary` on my compute, the test case is normal:

```
#[test]
fn test() {
    use crate::tester::Tester;

    let pass = vec![r#""x ? y";"#];

    let fail = vec![
        "var foo = true ? thing : stuff;",
        "true ? thing() : stuff();",
        "function foo(bar) { return bar ? baz : qux; }",
    ];

    Tester::new(NoTernary::NAME, pass, fail).test_and_snapshot();
}
```

Reply:
@mysteryven Maybe I haven't explained it correctly, the result of the rulegen is the same for you and me, as I said, I added the second test, the problem is that the first test that should pass fails.

Reply:
Sorry, I'm reviewing it and I have confused the error, the fail comes from other test

Reply:
Sorry to misunderstand, are you confused the by the [`r#` syntax](https://doc.rust-lang.org/rust-by-example/std/str.html#literals-and-escapes) 🤔, `r#""x ? y";"#` is same as `"x ? y"` in JavaScript.

Reply:
Now should be working, but I don't understand why I haven't to check if the conditional is a binary operator 🤔

Reply:
The `run` function will visit every AST node in the source text :)

https://github.com/oxc-project/oxc/blob/02c82c3f78faa8db832a519e953986fa430da0d6/crates/oxc_linter/src/lib.rs#L124-L130